src/test/kotlin/lotto/domain/LottoResultTest.kt

```diff
@@ -0,0 +1,57 @@
+package lotto.domain
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+class LottoResultTest {
+
+    @Test
+    fun `should calculate correct revenue with multiple prizes`() {
+        // given
+        val prizes = listOf(
+            Prize.FIFTH,  // 5_000
+            Prize.FOURTH, // 50_000
+            Prize.SECOND  // 30_000_000
+        )
+        val purchaseAmount = LottoPurchaseAmount(3_000) // 3 tickets
+
+        // when
+        val result = LottoResult.of(prizes, purchaseAmount)
+
+        // then
+        val expectedReward = Prize.FIFTH.reward + Prize.FOURTH.reward + Prize.SECOND.reward
+        val expectedRevenue = expectedReward.toDouble() / purchaseAmount.value * 100
+
+        assertThat(result.revenue).isEqualTo(expectedRevenue)
+        assertThat(result.prizes).containsExactlyElementsOf(prizes)
```

Comment:
```suggestion
        assert(result.revenue == expectedRevenue)
        assert(result.prizes == prizes)
```

suggestions, up to your preferences